src/main/client/src/components/@pages/ProfilePage.tsx

```diff
@@ -1,33 +1,104 @@
 import { Page } from '../@layout/Page'
-import { Text, Flex, Heading, CircularProgress, CircularProgressLabel, Wrap, WrapItem, Center } from '@chakra-ui/react'
+import {
+  Text,
+  Flex,
+  Heading,
+  CircularProgress,
+  CircularProgressLabel,
+  Wrap,
+  WrapItem,
+  Center,
+  Skeleton,
+  VStack,
+  SkeletonCircle,
+  Button,
+  HStack
+} from '@chakra-ui/react'
+import * as React from 'react'
+import { ProfileDTO, RoleType } from 'types/dto/profile'
+import axios from 'axios'
+import { API_BASE_URL } from 'utils/configurations'
 
-const challenges = [
+const challenges = (profile: ProfileDTO) => [
   {
     name: 'Bucketlist',
-    value: 40
+    value: profile?.submittedAchievementCount,
+    total: profile?.totalAchievementCount
   },
   {
     name: 'Riddle',
-    value: 73
+    value: profile?.completedRiddleCount,
+    total: profile?.totalRiddleCount
   },
   {
     name: 'QR kód',
-    value: 17
+    value: profile?.tokens?.length,
+    total: profile?.minTokenToComplete
   }
 ]
 
 type ProfilePageProps = {}
 
+//Had to create a seperate skeleton layout so it wouldn't look strange
+export const ProfilePageSkeleton: React.FC<ProfilePageProps> = (props) => {
+  return (
+    <Page {...props}>
+      <VStack align="flex-start" mb="14" mt={6}>
+        <Skeleton h={12} w={['40%', null, null, '15%']}></Skeleton>
+        <Skeleton h={10} w={['50%', null, null, '20%']}></Skeleton>
+        <Skeleton h={10} w={['60%', null, null, '25%']}></Skeleton>
+      </VStack>
+      <Wrap spacing="3rem" justify="center" mt={3}>
+        {[0, 1, 2].map((challenge) => (
+          <WrapItem key={challenge}>
+            <Center w="10rem" h="12rem">
+              <Flex direction="column" align="center">
+                <Skeleton mb={3} w="90%" h={10} />
+                <SkeletonCircle size="10rem" />
+              </Flex>
+            </Center>
+          </WrapItem>
+        ))}
+      </Wrap>
+    </Page>
+  )
+}
+
 export const ProfilePage: React.FC<ProfilePageProps> = (props) => {
+  const [profile, setProfile] = React.useState<ProfileDTO | undefined>(undefined)
+  const [delay, setDelay] = React.useState<boolean>(false)
+
+  React.useEffect(() => {
+    //delay is needed to remove flickering of the skeleton
+    setDelay(true)
```

Comment:
Why don't you wrap the API call into a timeout?

Reply:
That would cause the loading time to be delay + response time, this way it is Max(delay, response time). This seemed like a smoother solution. If you would like me to change it I'll remove the delay state.

Reply:
Oh okay, I see what you did here.